website/server/controllers/api-v3/members.js

```diff
@@ -270,6 +270,44 @@ api.getChallengeMemberProgress = {
 };
 
 /**
+ * @api {get} /api/v3/members/:toUserId/objections-to/:interaction Get the message of any errors that would occur if the given interaction was attempted
+ * @apiVersion 3.0.0
+ * @apiName GetObjectionsToInteractionIfAny
+ * @apiGroup Member
+ *
+ * @apiParam {UUID} toUserId The user to interact with
+ * @apiParam {String} interaction Name of the interaction to query, e.g. "send-private-message"
+ *
+ * @apiSuccess {Array} data Return an array of error messages, if the interaction would be blocked; otherwise an empty array
+ */
+api.getObjectionsToInteractionIfAny = {
+  method: 'GET',
+  url: '/members/:toUserId/objections-to/:interaction',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    req.checkParams('toUserId', res.t('toUserIDRequired')).notEmpty().isUUID();
+    req.checkParams('interaction', res.t('interactionRequired')).notEmpty();
+
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let sender = res.locals.user;
+    let receiver = await User.findById(req.params.toUserId).exec();
+    if (!receiver) throw new NotFound(res.t('userWithIDNotFound', {userId: req.params.toUserId}));
+
+    let response;
+
+    try {
+      response = sender.getObjectionsToInteractionIfAny(req.params.interaction, receiver);
+    } catch (e) {
+      throw new NotFound(e.message);
```

Comment:
The callee is throwing a normal `Error` instead of a HTTP `NotFound` to be independent of HTTP. Unfortunately, an unhandled `Error` will be turned into a generic 500 Internal Server Error and the message lost. This is fine for most API functions, but when the cause of the error can be an API parameter directly, I think the error message needs to be exposed to the user.